2007-11-12  Corentin Le Gall <clegall@proformatique.com>

        pf_queue_agents_language.patch
        $Revision$
        $Date$
        Bugfix / Behavioral change

        Propagates the language setting to the subsequently created channels,
        among which especially the agents.

        Patch status: a few tests (OK) have been done, need more statistics

Index: asterisk-1.2.24/channel.c
===================================================================
--- asterisk-1.2.24.orig/channel.c	2007-11-12 18:40:49.000000000 +0100
+++ asterisk-1.2.24/channel.c	2007-11-12 18:40:51.000000000 +0100
@@ -2896,6 +2896,7 @@
 	struct ast_var_t *current, *newvar;
 	char *varname;
 
+	ast_copy_string(child->language, parent->language, sizeof(child->language));
 	AST_LIST_TRAVERSE(&parent->varshead, current, entries) {
 		int vartype = 0;
 
